lib/events-reader.js

```diff
@@ -245,37 +229,18 @@ module.exports = function (harvesterApp) {
             }
         };
 
-        var checkpointUpdate = function(that,doc){
-            return harvesterApp.adapter.update('checkpoint', that.checkpoint.id, {ts: doc.ts})
-        };
-
-        var debouncedCheckpointUpdate = debounce(checkpointUpdate,parseInt(_.get(harvesterApp, 'options.eventsReaderDebounceWait'), 10) || 100,{leading: true});
-
         EventsReader.prototype.updateCheckpointAndReschedule = function (doc) {
             var that = this;
             if (doc != null) {
 
-                var regexProfileMessage = 'checkpoint regex to cl ' + doc.ns + ' ts ' +  doc.ts;
-                log.profile(regexProfileMessage);
                 var regexCheckpoint = new RegExp('.*\\.checkpoints$', 'i');
                 var matchCheckpoint = regexCheckpoint.test(doc.ns);
-                log.profile(regexProfileMessage);
 
                 if (!matchCheckpoint) {
-
-                    var profileMessage = 'updateCheckpointAndReshedule ' + doc.ts;
-
-                    debug('doc checkpoint ' + doc.ts, doc);
-                    log.profile(profileMessage);
                     debug('updating checkpoint with ts: ' + logTs(doc.ts));
+                    checkpointWriter.checkpointEvent.emit('newCheckpoint', that.checkpoint.id, doc);
```

Comment:
You should pull some of this internal; 'newCheckpoint' is tightly coupled to the writerloop functionality. Would be better to have a checkpointWriter.debounceCheckpointWrite(checkpoint.id,doc) or something similar